code/script.js

```diff
@@ -222,104 +176,177 @@ const generateBoard = () => {
+      charactersInPlay = charactersInPlay.filter((person) => person.hair !== value);
     }
   } else if (category === 'other') {
-    // Similar to the one above
-  } else {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
+      alert(`Yes, the person is ${value}! Keep all people that are ${value}`)
+      charactersInPlay = charactersInPlay.filter((person) => person.other.includes(value));
     } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+      alert(`No, the person is not ${value}! Remove all people that are ${value}`)
+      charactersInPlay = charactersInPlay.filter((person) => !person.other.includes(value));
     }
-  }
-
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
-
-  // Invoke a function to redraw the board with the remaining people.
+  } else if (category === 'eyes') {
+    if (keep) {
+      alert(`Yes, the person has ${value} eyes! Keep all people that has ${value} eyes`);
+      charactersInPlay = charactersInPlay.filter((person) => person.eyes === value);
+    } else {
+      alert(`No, the person doesn't have ${value} eyes! Remove all people that have ${value} eyes`);
+      charactersInPlay = charactersInPlay.filter((person) => person.eyes !== value);
+    }
+  }    
+  generateBoard();
 }
 
-// when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+  let confirmChoice = confirm('Sure you wanna guess? It will be the last draw. Please confirm');
+    if (confirmChoice === true) {
+      checkMyGuess(personToConfirm);
+      stopTimer();
+    } else {
+      console.log('pressed cancel')
+    }
+}
+
+let guessStatus = {
+  status: ''
 }
 
-// If you confirm, this function is invoked
 const checkMyGuess = (personToCheck) => {
-  // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
+    if (secret.name === personToCheck) {
+    guessStatus.status = 'right';
+  } else {
+    guessStatus.status = 'wrong';
+  }
+
+  board.innerHTML = ''
+
+  board.innerHTML += `
```

Comment:
The hover effect over the icon at the result view can't be removed on mobile view. When you click on the icon the hover effect still appears even when you reset the game.